### PR TITLE
CloudStack multiple worker node groups e2e test and e2e refacatoring

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
Refactoring CloudStack e2e tests to align closer with VSphere to support multiple worker node group tests.

*Testing (if applicable):*
MWNG E2E test passing locally

```
2022-04-18T16:28:54.173-0400	V4	Tasks completed	{"duration": "9m5.266951366s"}
2022-04-18T16:28:54.173-0400	V3	Cleaning up long running container	{"name": "eksa_1650313187367819000"}
2022-04-18T16:28:54.614-0400	V3	e2e	Cleaning up long running container	{"name": "eksa_1650312207227828000"}
--- PASS: TestCloudStackKubernetes121RedhatAndAndRemoveWorkerNodeGroups (1527.80s)
PASS
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/api/cloudstack.go [hunk 1]
```diff
@@ -4,105 +4,63 @@ import (
...
-	// only to get name of control plane and worker node machine configs
-	clusterConfig, err := v1alpha1.GetAndValidateClusterConfig(filename)
-	if err != nil {
-		return nil, fmt.Errorf("the cluster config file provided is invalid: %v", err)
-	}
-
-	cpName := clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
-	workerName := clusterConfig.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name
-	cloudstackDatacenterConfig, err := v1alpha1.GetCloudStackDatacenterConfig(filename)
+	cloudstackDatacenterConfig, err := anywherev1.GetCloudStackDatacenterConfig(filename)
```
Comment (jiayiwang7) [inline internal/pkg/api/cloudstack.go:22]: u can simply use `cluster.ParseConfigFromFile(filename)` to get the config obj that has everything

https://github.com/aws/eks-anywhere/blob/main/internal/pkg/api/snow.go#L22

File: test/e2e/multiple_worker_node_groups_test.go [hunk 2]
```diff
@@ -49,3 +50,43 @@ func TestVSphereKubernetes122BottlerocketAndAndRemoveWorkerNodeGroups(t *testing
...
 }
+
+func TestCloudStackKubernetes121RedhatAndAndRemoveWorkerNodeGroups(t *testing.T) {
+	provider := framework.NewCloudStack(t,
+		framework.WithCloudStackWorkerNodeGroup(
+			"worker-1",
+			framework.WithWorkerNodeGroup("workers-1", api.WithCount(2)),
+		),
+		framework.WithCloudStackWorkerNodeGroup(
+			"worker-1",
```
Comment (jiayiwang7) [inline test/e2e/multiple_worker_node_groups_test.go:61]: "worker-2" ?

File: test/framework/cloudstack.go [hunk 3]
```diff
@@ -168,13 +174,45 @@ func (c *CloudStack) ClusterConfigFillers() []api.ClusterFiller {
...
 }
+
+func (c *CloudStack) WithNewCloudStackWorkerNodeGroup(name string, workerNodeGroup *WorkerNodeGroup, fillers ...api.CloudStackMachineConfigFiller) ClusterE2ETestOpt {
+	return func(e *ClusterE2ETest) {
+		e.ProviderConfigB = c.customizeProviderConfig(e.ClusterConfigLocation, cloudStackMachineConfig(name, fillers...))
+		var err error
+		// Using the ClusterConfigB instead of file in disk since it might have already been updated but not written to disk
+		e.ClusterConfigB, err = api.AutoFillClusterFromYaml(e.ClusterConfigB, buildCloudStackWorkerNodeGroupClusterFiller(name, workerNodeGroup))
+		if err != nil {
+			e.T.Fatalf("Error filling cluster config: %v", err)
```
Comment (jiayiwang7) [inline test/framework/cloudstack.go:195]: nit
```suggestion
			e.T.Fatalf("filling cluster config: %v", err)
```

File: test/framework/cloudstack.go [hunk 4]
```diff
@@ -168,13 +174,45 @@ func (c *CloudStack) ClusterConfigFillers() []api.ClusterFiller {
...
+		}
+	}
+}
+
+func cloudStackMachineConfig(name string, fillers ...api.CloudStackMachineConfigFiller) api.CloudStackFiller {
+	f := make([]api.CloudStackMachineConfigFiller, 0, len(fillers)+1)
```
Comment (jiayiwang7) [inline test/framework/cloudstack.go:201]: shouldn't this be
```suggestion
	f := make([]api.CloudStackMachineConfigFiller, 0, len(fillers)+2)
```

File: internal/pkg/api/cloudstack.go (hunk 1 above)
Comment (maxdrib) [reply internal/pkg/api/cloudstack.go:22]: works for me! Thanks

File: test/e2e/multiple_worker_node_groups_test.go [hunk 5]
```diff
@@ -49,3 +50,43 @@ func TestVSphereKubernetes122BottlerocketAndAndRemoveWorkerNodeGroups(t *testing
 		),
 	)
 }
+
+func TestCloudStackKubernetes121RedhatAndAndRemoveWorkerNodeGroups(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (maxdrib) [reply test/e2e/multiple_worker_node_groups_test.go:54]: Good catch

File: test/e2e/multiple_worker_node_groups_test.go (hunk 2 above)
Comment (maxdrib) [reply test/e2e/multiple_worker_node_groups_test.go:61]: very good catch

File: test/framework/cloudstack.go (hunk 3 above)
Comment (maxdrib) [reply test/framework/cloudstack.go:195]: Fixed

File: test/framework/cloudstack.go (hunk 4 above)
Comment (maxdrib) [reply test/framework/cloudstack.go:201]: Fixed, thanks

**Review by vignesh-goutham — APPROVED** [review]

**Pull request merged**